Formula/catboost.rb

```diff
@@ -0,0 +1,24 @@
+class Catboost < Formula
+  desc "Fast, scalable, high performance Gradient Boosting on Decision Trees library"
+  homepage "https://catboost.ai"
+  url "https://github.com/catboost/catboost/archive/v0.24.4.tar.gz"
+  sha256 "03df498249206519b7e37ff8067cf265a2826afc4305a32f97986fbe308994da"
+  license "Apache-2.0"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  def install
+    cd "#{buildpath}/catboost/app" do
+      ENV["YA_CACHE_DIR"] = "./.ya"
+      system "../../ya", "make", "-r", "-o", "#{buildpath}/build"
```

Comment:
`ya` looks like it downloads something. I think that should be declared as a `resource` instead.

Reply:
Actually, it looks like it downloads a binary executable. We should build that from source instead.

Reply:
OK!

I'll try to figure out how to do it!

Reply:
It looks like `ya-bin` is not open source as explained here: https://github.com/catboost/catboost/issues/411#issuecomment-404250907

They provide makefiles to build catboost without `ya-bin`, but there is not makefile for macOS.

Can we continue building the formula as it is now?

We can create makefiles manually with `ya make --makefile` and add them as patches, but we need a makefile for x86_64 and Apple Silicon architectures.

Reply:
If it pulls in closed source dependencies this is not a good fit for homebrew-core.

Please consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

Reply:
OK!

No problem, I already has my own tap for these cases.

I'll open a new PR if the catboost team change the way they compile it source code.